ome_zarr/writer.py

```diff
@@ -172,28 +198,39 @@ def write_multiscale(
     axes: str or list of str or list of dict
       List of axes dicts, or names. Not needed for v0.1 or v0.2
       or if 2D. Otherwise this must be provided
-    transformations: 2Dlist of dict
-      For each path, we have a List of transformation Dicts (not validated).
-      Each list of dicts are added to each datasets in order.
+    coordinate_transformations: 2Dlist of dict
+      For each path, we have a List of transformation Dicts.
+      Each list of dicts are added to each datasets in order
+      and must include a 'scale' transform.
     """
 
     dims = len(pyramid[0].shape)
     axes = _get_valid_axes(dims, axes, fmt)
 
-    paths = []
-    for path, dataset in enumerate(pyramid):
+    datasets: List[dict] = []
+    for path, data in enumerate(pyramid):
         # TODO: chunks here could be different per layer
-        group.create_dataset(str(path), data=dataset, chunks=chunks)
-        paths.append(str(path))
-    write_multiscales_metadata(group, paths, fmt, axes, transformations)
+        group.create_dataset(str(path), data=data, chunks=chunks)
+        datasets.append({"path": str(path)})
+
+    shapes = [data.shape for data in pyramid]
+    if coordinate_transformations is None:
+        coordinate_transformations = fmt.generate_coordinate_transformations(shapes)
+    fmt.validate_coordinate_transformations(
```

Comment:
Although this does not hurt, is that call now redundant with the one that is called later during the validation of the `datasets` element?

Reply:
Not quite! If you pass in a list of transformations that is longer than the number of pyramid levels, this will fail validation at this point. BUT, if we remove the validation here it *won't* fail later because the transformations are zipped onto the Datasets, so the lengths will match (and the extra transformations will be ignored).
It is probably better to fail than to ignore the mismatch.

Reply:
👍 Semi-related I discovered that Python 3.10 introduced a `strict` argument to [zip](https://docs.python.org/3.10/library/functions.html#zip) that will eventually allow similar code to fail on mismatching list lengths rather than truncating. Let's keep things as they are for now but maybe add an inline comment explaining this logic?

Reply:
Done in [b081342](https://github.com/ome/ome-zarr-py/pull/162/commits/b08134238d9552b2c46b821ff6effb4cd958a9e0)